src/main/java/org/apache/sysml/runtime/controlprogram/ProgramBlock.java

```diff
@@ -363,7 +368,16 @@ private static void checkSparsity( Instruction lastInst, LocalVariableMap vars )
 					synchronized( mb ) { //potential state change
 						mb.recomputeNonZeros();
 						mb.examSparsity();
+
 					}
+					if(sparse1) {
```

Comment:
Note that checking sparse1 might lead to NPEs because the previous call to examSparsity might change the representation from sparse to dense. Please check mb.isInSparseFormat instead.